File: api/market/overview.js
import { fetchAllStocks } from '../../lib/stock-data.js';
import { getMarketOverview } from '../../lib/analysis.js';

export default async function handler(req, res) {
  try {
    const stocks = await fetchAllStocks();
    const overview = getMarketOverview(stocks);
    overview.time = new Date().toLocaleString('zh-CN', { timeZone: 'Asia/Shanghai' });

    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Cache-Control', 's-maxage=30');
    res.status(200).json({ ok: true, data: overview });
  } catch (e) {
    res.status(500).json({ ok: false, detail: e.message });
  }
}

File: api/market/overview.py
from http.server import BaseHTTPRequestHandler
import json
import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))

from app.services.stock_data import get_all_stocks
from app.services.analysis import get_market_overview
from datetime import datetime


class handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

    def do_GET(self):
        try:
            df = get_all_stocks()
            overview = get_market_overview(df)
            overview["time"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
            self._respond(200, {"ok": True, "data": overview})
        except Exception as e:
            self._respond(500, {"ok": False, "detail": str(e)})

    def _respond(self, code, data):
        self.send_response(code)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        self.wfile.write(json.dumps(data, ensure_ascii=False).encode())
